contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -993,7 +993,7 @@ abstract contract AbstractTokenConverter is AccessControlledV8, IAbstractTokenCo
 
                 uint256 balanceBefore = IERC20Upgradeable(tokenAddressIn).balanceOf(_destinationAddress);
 
-                IERC20Upgradeable(tokenAddressOut).approve(converterAddresses[i], amountIn);
+                IERC20Upgradeable(tokenAddressOut).forceApprove(converterAddresses[i], amountIn);
```

Comment:
Don't we need first approve 0 ?

Reply:
The function forceApprove checks if allowance is zero or not. If it is non-zero it first sets it to zero then to the desired amount. see [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/442886ed5ff8a0b9ab477b191f5238541ee6d772/contracts/token/ERC20/utils/SafeERC20.sol#L74).
So, we do not need to set it first to zero.